pkg/proxy/iptables/proxier.go

```diff
@@ -1150,6 +1152,8 @@ func (proxier *Proxier) syncProxyRules() {
 						utilproxy.WriteLine(proxier.natRules, append(args, "-j", string(KubeMarkMasqChain))...)
 					}
 				}
+				// Changing the chain name (args[1]) to which we append the rule.
+				args[1] = string(kubeServicesChain)
```

Comment:
[EDIT: the comment below about `utilproxy.WriteRuleLine` might make this comment irrelevant]

Changing one element of `args` is a little confusing, and fragile... I think it might be better to just duplicate the whole `args = ...`

Also, it's weird that we initially create `args` as it would be used for the optional `!svcInfo.NodeLocalExternal()` case, and then afterward update it for the non-optional case, possibly having never used the `-A svcChain` version... But you should be able to move the optional part to after the non-optional part now; since they're appending to different chains now, it doesn't matter which order you write the lines in. Though be careful about `destChain`...